### PR TITLE
Update readme

### LINKED ISSUE #1: Binary for Intel-based Macs
**Background:**
During our first developer onboarding call, the interview participant was unable to launch a REPL because we didn't include a binary for older Mac's. We started to build the project, but Nix wasn't installed on his personal machine and wasn't finished by the time the call was done.

**Result:**
Going through the docs was still useful, but we lost the impact of running the examples locally and gaining hands on experience with Sire. 

**Proposal:**
Create a binary for intel-based Macs.

Binaries shouldn't be relied on in general, but are the fastest way to get the system running right now. Time-to-REPL is the best proxy we have for initial onboarding friction.

### LINKED ISSUE #4: What’s cool here; wow moments/claims

### LINKED ISSUE #6: Make a point about this being a language platform?

### LINKED ISSUE #7: Improve the caveats section so that it doesn’t immediately contradict claims made in the intro (or re-word those claims)

### PR DESCRIPTION
**Pull request by vcavallo**
Here is the updated Readme for final review.
Closes #7
Closes #6 
Closes #4 
Closes #1

### CODE REVIEW

**Comment by vcavallo** [comment]
@kjekac @cvanetten One thing I want to call out as maybe-odd: in the readme, we should a very terse version of the count-up cog (see screenshot below)

image

But this PR proposes using a _very_ verbosely-commented version of the same file (see files changed to review). The readme does mention that the actual source file has a lot more comments to peruse. Is this "conflict" too confusing? I think it's fair to have a _screenshot_ in the readme with a very terse explanation and then a pointer to the actual file with more comments, but I can see it being confusing maybe.

**Comment by cvanetten** [comment]
I don't think it's confusing. It's not ideal to have a screenshot instead of a block of code, but we already discussed that trade off and I agree this is reasonable for now

File: README.md [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
+6. [Caveats](#caveats)
 7. [Additional Resources](#additional-resources)
 
 ## Introduction
 
+Pallas is an event sourced, purely functional operating system, called an operating function. 
 
-Pallas is a purely functional exokernel and library OS, designed to radically simplify the modern networked computing stack. We call the libOS an "operating function" because it is defined as a pure function of its event input stream.
-The Pallas [SSI](https://wiki.vaporware.network/solid-state%20interpreter) programming environment is a virtual machine which is also a database engine. Databases are created by writing pure functions which fold their streams of incoming events, at each step returning an output event and possibly an upgraded version of the function itself.
+Pallas provides an entirely unique set of features, but is inspired by a long history of systems and language research, including MIT’s exokernel. Pallas collapses the distinction between database, virtual machine, and language platform, creating a stable, simple, and extensible application environment.
```
Comment (kjekac) [inline README.md:19]:
```suggestion
Pallas provides an entirely unique set of features, but is inspired by a long history of systems and language research, including MIT’s exokernel. Pallas collapses the distinction between database, virtual machine, and language platform, creating a stable, simple, and extensible programming environment.
```

File: README.md [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
 
-Let's pick that apart. The foundation of Pallas is untyped, but conceptually we can say that a database is a function of the type `DB`:
+The cost to this design is a clean break with legacy systems, including Unix and most popular programming languages. In return, Pallas offers a set of features that are found nowhere else:
+- All application code is automatically persisted, without imports or boilerplate. To create a database, you write a pure function. System calls are included in persisted state.
```
Comment (kjekac) [inline README.md:22]:
```suggestion
- All application code is automatically persisted, without imports or boilerplate. To create a database, you write a pure function. System calls are included in persisted state, and so are restarted on reboot.
```

File: README.md [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
+- Partially applied functions can be serialized and stored on-disk, or sent over the wire. Programs in mid-execution can be paused, moved to a new machine, and resumed with no impact.
+- Program execution can be parallelized via a process model. A single machine can spawn and manage multiple processes.
+- The system’s language blends features from both Lisp and Haskell, but uses a more readable and flexible syntax than S-expressions. Pallas supports metaprogramming, hot reload, and macro-based type systems. 
+- Data and code is deduplicated, merkleized, and stored in content-addressable memory pages. This creates a global referentially-transparent content store, which naturally complements protocols like BitTorrent.
+
+By reading our documentation and examples, you can prove to yourself that these claims are true.
```
Comment (kjekac) [inline README.md:28]:
```suggestion
```

Comment (kjekac) [reply README.md:28]: Feels insecure? Maybe we should keep it but say something more like "There are [examples] and [docs] here, and if you want to talk to us we have a [Telegram]." Basically treat this as the end of the intro, and then launch into the more technical stuff below.

**Review by kjekac — APPROVED** [review]
Looks good generally imo. A few suggestions here!

**Comment by kjekac** [comment]
Reading this now, i think a major thing i would like to see as a reader is a brief overview of the architecture. As-is, we jump straight from "this is what this is on a high level" into "here's Sire".

"Okay, this is an event-sourced operating function, cool, i can kind of make sense of that if i think about it. I wonder what it consists of, does the design make sense?"
"Here's how you write apps in it!"

We don't have to go deep into how we do IO, but just something like the [tower diagram i showed at LC](https://docs.google.com/presentation/d/1Sgdy3ZakZV-SuER-afLBqlzl6rXnJHMSnWQJ4LH3yTU/edit#slide=id.g2d9f0dcf67c_0_176) and maybe say something about
- Sire bootstrapping itself from PLAN
- Rex
- (that the runtime acts as a kernel outside the system ("exokernel") which interprets calls to hardware completely opaquely, thus allowing the formal system state to be decoupled from any particular implementation).

Should i create a new issue for this? I think this PR is fine to merge as-is since it's a clear improvement.

File: README.md (hunk 2 above)
Comment (kjekac) [reply README.md:22]: Maybe unnecessary but it seems reasonable to spell out why this is important if we're gonna mention it at all.

**Comment by kjekac** [comment]
Btw i don't think this closes #1? You might want to check your list there.

**Comment by vcavallo** [comment]
> Btw i don't think this closes #1? You might want to check your list there.

It does because I've uploaded the binary and added the link to it here.
But I'm still testing the intel binary (long preamble on this old mac) so it's a bit presumptuous

**Comment by vcavallo** [comment]
> Reading this now, i think a major thing i would like to see as a reader is a brief overview of the architecture. As-is, we jump straight from "this is what this is on a high level" into "here's Sire".
> 
> "Okay, this is an event-sourced operating function, cool, i can kind of make sense of that if i think about it. I wonder what it consists of, does the design make sense?" "Here's how you write apps in it!"
> 
> We don't have to go deep into how we do IO, but just something like the [tower diagram i showed at LC](https://docs.google.com/presentation/d/1Sgdy3ZakZV-SuER-afLBqlzl6rXnJHMSnWQJ4LH3yTU/edit#slide=id.g2d9f0dcf67c_0_176) and maybe say something about
> 
>     * Sire bootstrapping itself from PLAN
> 
>     * Rex
> 
>     * (that the runtime acts as a kernel outside the system ("exokernel") which interprets calls to hardware completely opaquely, thus allowing the formal system state to be decoupled from any particular implementation).
> 
> 
> Should i create a new issue for this? I think this PR is fine to merge as-is since it's a clear improvement.

plopping this updated tower diagram here for use later:

image

**Pull request merged**